lib/iris/fileformats/pp.py

```diff
@@ -2140,6 +2141,54 @@ def load_cubes(filenames, callback=None, constraints=None):
                                        constraints=constraints)
 
 
+def load_pairs_from_fields(pp_fields):
+    """
+    Convert an iterable of PP fields into an iterable of tuples of
+    (Cubes, PPField).
+
+    Args:
+
+    * pp_fields:
+        An iterable of :class:`iris.fileformats.pp.PPField`.
+
+    Returns:
+        An iterable of :class:`iris.cube.Cube`s.
+
+    This capability can be used to filter out fields before they are passed to
+    the load pipeline, and amend the cubes once they are created, using
+    PP metadata conditions.  Where this filtering
+    removes a significant number of fields, the speed up to load can be
+    significant:
+
+        >>> import iris
+        >>> from iris.fileformats.pp import load_pairs_from_fields
+        >>> filename = iris.sample_data_path('E1.2098.pp')
+        >>> filtered_fields = []
+        >>> for field in iris.fileformats.pp.load(filename):
+        ...     if field.lbproc == 128:
+        ...         filtered_fields.append(field)
+        >>> cube_fields = load_pairs_from_fields(filtered_fields)
```

Comment:
Could this instead read:

`cube_field_pairs = load_pairs_from_fields(filtered_fields)`

? I was temporarily confused by what `cube_fields` was.